GSoC-2025/cellfinder: Improving cellfinder's classification algorithm (Srutanik).md

```diff
@@ -0,0 +1,111 @@
+      * **(Stretch Goal) Ensemble Model:** Implementation of an ensemble method (e.g., weighted averaging of predictions) combining the outputs of the best-performing newly implemented models (e.g., Attention-based + WaveMix) to potentially maximize overall classification accuracy and robustness.
+
+   2. **Analysis, Documentation, and Tests:**
+      * **Comparative Analysis:** Analysis of the implemented architectures against the ResNet-50 baseline, using standard classification metrics (F1 score, sensitivity/recall, specificity, accuracy, AUC, loss curves) and computational performance metrics (inference time, training time, GPU memory usage in flops).
+      * **Explainable Model Outputs:** Implementation of Grad-CAM and Guided Grad-CAM techniques to provide visual explanations of model decisions, highlighting regions in the 3D volumes that contributed most to classification outcomes with the help of saliency maps.
+      ([Ref Paper Link](https://arxiv.org/abs/1610.02391))
+      * **Documentation & Blog Post:** Comprehensive documentation integrated into `cellfinder`'s existing documentation, detailing the new architectures, the rationale for their use, instructions for usage (training, inference), configuration options, and a summary of the comparative analysis findings including advantages and disadvantages of each technique in the `cellfinder` context. A blog post will be written to communicate the project's goals, methods, and key results.
+      * **Testing:** Robust test suite including unit tests for core functions within the new model implementations (e.g., custom layers, loss functions) and integration tests to ensure the new models integrate correctly with `cellfinder`'s existing data loading, preprocessing, training, and inference pipelines.
+
+### Implementation Timeline
+
+* **Minimal Set of Deliverables:**
+  1. Model Implementations: **3D CNN + Attention Gate** and **3D WaveMix Adaptation** (first two bullet points under deliverable 1).
+  2. Analysis, Documentation, and Tests: All items under deliverable 2 covering the implemented models and baseline, including Grad-CAM/Guided Grad-CAM implementations.
+
+* **Stretch Goals:**
+  1. Model Implementation: **3D PIP-Net Adaptation** (third bullet point under deliverable 1).
+  2. Model Implementation: **Ensemble Model** (fourth bullet point under deliverable 1).
+
+* **Weekly Timeline** (Approx. 40 hours/week)
+
+| **Dates** | **GSoC Phase** | **Key Tasks & Milestones** |
+|-----------|----------------|----------------------------|
+| May 8 – June 1 | **Community Bonding** | - Finalize architecture details (backbones, attention type, WaveMix specifics) with mentors.<br>- Deep dive into `cellfinder` classification codebase & data handling.<br>- Set up environment & baseline ResNet benchmarking framework. |
```

Comment:
Rather than deciding on the architecture at the beginning, would it be better to trial a few prototypes?